distribution/errors.go

```diff
@@ -78,7 +78,7 @@ func TranslatePullError(err error, ref reference.Named) error {
 		switch v.Code {
 		case errcode.ErrorCodeDenied:
 			// ErrorCodeDenied is used when access to the repository was denied
-			newErr = errors.Errorf("repository %s not found: does not exist or no pull access", reference.FamiliarName(ref))
+			newErr = errors.Errorf("pull access denied for %s, repository does not exist or may require 'docker login'", reference.FamiliarName(ref))
```

Comment:
The nasty matching is here; https://github.com/moby/moby/blob/6486924f0b6464a90c9c2d251a3ab0a96302dce9/api/server/httputils/errors.go#L56-L68 looks like it's not depending on these 👍